test/sql/copy/per_thread_output.test

```diff
@@ -56,10 +56,10 @@ SELECT COUNT(*) FROM PARQUET_SCAN('__TEST_DIR__/per_thread_output2/*.parquet')
 10000
 
 
-# cant write to existing folders
 statement error
 COPY (FROM bigdata) TO '__TEST_DIR__/per_thread_output2/' (FORMAT PARQUET, PER_THREAD_OUTPUT TRUE);
 ----
+IO Error
```

Comment:
I thought about that, but the downside of being so specific is that tests have to be updated when the error message changes
If we're fine with that though then I would prefer them to be more specific too

Reply:
I think if you are changing an error message, then you probably have a reason for that. I.e., you should update the tests, also to see if it still makes sense that they fail. 🤔 Also, if we do not put a specific message, then we do not know if it is the correct `IO Exception`, etc.

Reply:
I'm for the more generic error messages. I think of it as checking the error type, but not the specific error message. If the IO Error message changes, then the IO Error type still applies. But if a different type of error message is thrown, then the test case should change.

With specific error messages you can also have downstream effects that are not immediately apparent which may also need to be fixed. Think about how the Nightly tests fail and then don't get fixed for a while. I know it's not the same thing, but I feel like the concept applies

Reply:
It's much more helpful when it shows specific error message, but it can make more tests false-fail in case the error messages change. Is there a way to check the type of the error and at the same time to compare the actual error message with the expected one? Maybe @Tmonster you know?

Reply:
Hmm, but generic error messages don't really help as much with regressions. Think of the binder. You can have a ton of different specific messages, and they are all just binder exceptions. E.g., in a test, you _should_ be able to qualify a column reference and fail somewhere later in the binding stack. But we'll never detect that now, as both will be binder exceptions.

W.r.t @hmeriann's question. To the best of my knowledge, the `SQLLogicTestRunner` currently does not support that distinction. If we want that distinction to both allow more robust tests (as highlighted by @Tmonster) and more specific tests for more regression-prone scenarios, then we need to allow different types of `statement error` types. E.g., something like `statement specific-error` (must match the text), `statement error-class` (must only match the error class), and `statement error` (like we have now, can match either). Not sure if adding these is too much, though.

Reply:
And, to add to your CI scenario. If a test is that wonky and just breaks while it should not (false fail), then that can also be an indication that the test should be written differently, e.g., in a more robust way.

Reply:
Some more context on this issue, I had this idea a while ago, but since it changed 800 files, we opted to not do it
I took the approach of making it as explicit as possible (that was also the easiest to script) and make it less explicit where needed (some tests behave differently on different platforms, in which case we can't be too specific)

<https://github.com/duckdb/duckdb/pull/9554>